sdk/PrebidDemandSDKLoaders/DemandSDKLoaders/FacebookAudienceNetwork/PBFacebookInterstitialAdLoader.m

```diff
@@ -0,0 +1,88 @@
+ limitations under the License.
+ */
+
+#import "PBFacebookInterstitialAdLoader.h"
+#import "PBConstants.h"
+
+@implementation PBFacebookInterstitialAdLoader
+
+- (void)loadInterstitialAd:(NSDictionary *)info {
+	// TODO nicole remove bid payload override
+	NSString *bidPayload = @"{\"type\":\"ID\",\"bid_id\":\"3657125005700902641\",\"placement_id\":\"1995257847363113_1997038003851764\",\"resolved_placement_id\":\"1995257847363113_1997038003851764\",\"sdk_version\":\"4.25.0-appnexus.bidding\",\"device_id\":\"87ECBA49-908A-428F-9DE7-4B9CED4F486C\",\"template\":102,\"payload\":\"null\"}";
+
+	// TODO nicole add this back in
+	//NSString *bidPayload = (NSString *)info[@"adm"];
+
+	// Load FBInterstitialAd using reflection so we can load the ad properly in the FBAudienceNetwork SDK
+	Class fbInterstitialAdClass = NSClassFromString(kFBInterstitialAdClassName);
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+	if (fbInterstitialAdClass != nil) {
+        SEL initMethodSel = NSSelectorFromString(kFBInterstitialInitMethodSelName);
+        id fbInterstitialAdObj = [fbInterstitialAdClass alloc];
+        if ([fbInterstitialAdObj respondsToSelector:initMethodSel]) {
+            NSString *placementId = [self parsePlacementIdFromBidPayload:bidPayload];
+            [fbInterstitialAdObj performSelector:initMethodSel withObject:placementId];
+
+            // Set selector variables for other methods we need to call on FBAdView
+            SEL setDelegateSel = NSSelectorFromString(kFBSetDelegateSelName);
+            SEL loadAdSel = NSSelectorFromString(kFBLoadAdWithBidPayloadSelName);
+
+            if ([fbInterstitialAdObj respondsToSelector:setDelegateSel] &&
+                [fbInterstitialAdObj respondsToSelector:loadAdSel]) {
+                    // Set up FBAdView and loadAdWithBidPayload
+                    [fbInterstitialAdObj performSelector:setDelegateSel withObject:self];
+                    [fbInterstitialAdObj performSelector:loadAdSel withObject:bidPayload];
+            }
+            self.interstitialAd = fbInterstitialAdObj;
+        }
+	}
+#pragma clang diagnostic pop
+}
+
+- (NSString *)parsePlacementIdFromBidPayload:(NSString *)bidPayload {
+	NSError *jsonError;
+	NSData *objectData = [bidPayload dataUsingEncoding:NSUTF8StringEncoding];
+	NSDictionary *json = [NSJSONSerialization JSONObjectWithData:objectData
+                                                         options:NSJSONReadingMutableContainers
+                                                           error:&jsonError];
+	return [json objectForKey:@"placement_id"];
+}
+
+- (void)showAdFromRootViewController:(UIViewController *)vc {
+	[self.interstitialAd showAdFromRootViewController:vc];
+}
+
+
+#pragma mark FBInterstitialAdDelegate methods
+
+- (void)interstitialAdDidLoad:(id)interstitialAd {
+	//[self.interstitialAd showAdFromRootViewController:[[UIApplication sharedApplication] keyWindow].rootViewController];
```

Comment:
Still curious why this does not depend audience network framework

Reply:
Since I initialize the empty protocol with the name FBInterstitialAdDelegate in the .h file, the compiler recognizes the protocol implemented in this file. Then at runtime when FB's interstitial calls the protocol, it calls the method because PBFacebookInterstitialAdLoader implemented the protocol and all of the methods its expecting. 

Let me know if that still doesn't make sense and we can chat about it offline too.